datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/ser/OneBasedMonthSerializer.java

```diff
@@ -0,0 +1,32 @@
+package com.fasterxml.jackson.datatype.jsr310.ser;
+
+import java.io.IOException;
+import java.time.Month;
+import java.util.function.Supplier;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+/**
+ * @since 2.17
+ */
+public class OneBasedMonthSerializer extends JsonSerializer<Enum<Month>> {
+    private final Supplier<Boolean> _serializeEnumsByIndex;
+    private final JsonSerializer<Enum> _defaultSerializer;
+
+    public OneBasedMonthSerializer(JsonSerializer<Enum> defaultSerializer, Supplier<Boolean> serializeEnumsByIndex) {
+        _defaultSerializer = defaultSerializer;
+        _serializeEnumsByIndex = serializeEnumsByIndex;
+    }
+
+    @Override
+    public void serialize(Enum<Month> value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+        if (_serializeEnumsByIndex.get()) {
```

Comment:
Use `SerializerProvider.isEnabled(SerializationFeature.WRITE_ENUMS_USING_INDEX)` instead of passing `Supplier`.

Reply:
Actually if/when serializer only registered when feature is enabled, can drop the check altogether.